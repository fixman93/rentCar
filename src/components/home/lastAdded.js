import React, { Component } from 'react'
import LastAddedItem from './lastAddedItem'

class LastAdded extends Component {

  render() {

    return (
      <section className='last-added'>
        <LastAddedItem />
        <LastAddedItem />
        <LastAddedItem />
        <LastAddedItem />
      </section>
    )
  }
}


export default LastAdded