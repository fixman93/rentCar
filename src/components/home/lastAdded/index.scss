.last-added {
  padding: 50px 0;
  // background-color: #f3f3f3;
  // background-size: cover;
}

.last-added-item {
  margin-bottom: 30px;
  position: relative;
  overflow: hidden;
  a {
    &:hover img {
      -webkit-transform: scale(1);
      transform: scale(1);
    }
  }
  img {
    max-width: 100%;
    transform: scale(1.04);
    transition: all .3s ease-in-out;
    height: 160px;
    object-fit: cover;
    width: 100%;
    object-position: center;
  }
  h3 {
    letter-spacing: 0;
    font-family: 'Frank Ruhl Libre', serif;
    margin-top: 10px;
    font-size: 24px;
    font-weight: bold;
    line-height: 1;
    span {
      font-size: 16px;
      margin-left: 8px;
    }
  }
  p {
    color: #9b9b9b;
    font-size: 15px;
    font-weight: bold;
  }
}

.vihicle-price {
  position: absolute;
  right: 0;
  bottom: 0;
  background: white;
  border-top-left-radius: 10px;
  padding: 5px 10px 25px 40px;
}

.vihicle-price .currency {
  font-size: 12px;
  text-align: left;
  position: absolute;
  left: 10px;
  top: 0;
}

.vihicle-price .price {
  font-size: 40px;
  font-family: 'Frank Ruhl Libre', serif;
  line-height: 29px;
  font-weight: bold;
}

.vihicle-price span {
  font-size: 13px;
  display: block;
  text-align: center;
}

section.last-added h2.title-block {
  text-align: left;
  font-size: 20px;
  margin-bottom: 10px;
  color: #999;
}