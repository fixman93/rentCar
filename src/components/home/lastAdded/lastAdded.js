import React, { Component } from 'react'
import Container from 'react-bootstrap/Container'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'

import LastAddedItem from './lastAddedItem'

import './index.scss'
export class LastAdded extends Component {

  renderItems = () => {
    const { projects } = this.props
    return projects && projects.map((project, i) => {
      return (
        <Col sm={4} key={i}><LastAddedItem className='LastAddedItem' project={project} /></Col>
      )
    })

  }
  render() {

    return (
      <section className='last-added'  >
        <Container>
          <h2 className='title-block text-gray'>You might like:</h2>
          <Row>
            {this.renderItems()}
          </Row>
        </Container>
      </section>
    )
  }
}


export default LastAdded