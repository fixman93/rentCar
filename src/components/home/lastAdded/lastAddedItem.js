import React, { Component } from 'react'
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom'

export class LastAddedItem extends Component {

  render() {
    const { project } = this.props
    console.log('eeeeee', project)
    return (
      <section className='last-added-item'>
        {project && project.id ?
          <Link to={'/project/' + project.id} >
            <img src={project.userImage} alt='Car' />
          </Link> : null}
        {project && project.carType ? <h3>{project.carType} {project.carModel} <span>{project.carYear}</span></h3> : null}
        <div className='vihicle-price'>
          {project && project.Currency ?
            <div className='currency'>{project.Currency}</div> :
            null}
          {project && project.carPrice ?
            <div className='price'>{project.carPrice}</div> :
            null}
          <span>per day</span>
        </div>
      </section>
    )
  }
}

LastAddedItem.propTypes = {
  project: PropTypes.object.isRequired
};


export default LastAddedItem 