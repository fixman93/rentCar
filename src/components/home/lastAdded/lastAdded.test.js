import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

import { LastAdded } from './lastAdded'

configure({ adapter: new Adapter() });

describe('LastAdded', () => {
  // const props = { projects: {} }
  const lastAdded = shallow(<LastAdded />)
  console.log('wwwwww', lastAdded.debug());
  it('render LastAdded', () => {
    expect(lastAdded).toMatchSnapshot()
  })

})