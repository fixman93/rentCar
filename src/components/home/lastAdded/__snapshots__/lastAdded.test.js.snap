// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastAdded render LastAdded 1`] = `ShallowWrapper {}`;
